### PR TITLE
add joypad keyboard

### PR DESCRIPTION
**Pull request by swiftrax**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by swiftrax** [comment]
image

File: keyboards/handwired/swiftrax/joypad/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Joypad", 
+    "url": "https://github.com/swiftrax", 
+    "maintainer": "swiftrax", 
+    "width": 4, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/info.json:8]:
```suggestion
        "LAYOUT_ortho_6x4": {
```

File: keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c:20]:
```suggestion
[0] = LAYOUT_ortho_6x4(
```

File: keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  KC_MUTE, KC_F13 , KC_F14 , KC_F15 ,
+  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+  KC_P7  , KC_P8  , KC_P9  , KC_PPLS,
+  KC_P4  , KC_P5  , KC_P6  , KC_PPLS,
+  KC_P1  , KC_P2  , KC_P3  , KC_PENT,
+  KC_P0  , KC_P0  , KC_PDOT, KC_PENT),
+
+[1] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c:28]:
```suggestion
[1] = LAYOUT_ortho_6x4(
```

File: keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+[2] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c:36]:
```suggestion
[2] = LAYOUT_ortho_6x4(
```

File: keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c:20]:
```suggestion
[0] = LAYOUT_ortho_6x4(
```

File: keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  KC_MUTE, KC_F13 , KC_F14 , KC_F15 ,
+  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+  KC_P7  , KC_P8  , KC_P9  , KC_PPLS,
+  KC_P4  , KC_P5  , KC_P6  , KC_PPLS,
+  KC_P1  , KC_P2  , KC_P3  , KC_PENT,
+  KC_P0  , KC_P0  , KC_PDOT, KC_PENT),
+
+[1] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c:28]:
```suggestion
[1] = LAYOUT_ortho_6x4(
```

File: keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+[2] = LAYOUT(
```
Comment (drashna) [inline keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c:36]:
```suggestion
[2] = LAYOUT_ortho_6x4(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/swiftrax/joypad/joypad.h [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+// readability
+#define XXX KC_NO
+
```
Comment (zvecr) [inline keyboards/handwired/swiftrax/joypad/joypad.h:22]:
```suggestion

```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  if(clockwise)
+    tap_code(KC_VOLU);
+  else
+    tap_code(KC_VOLD);
+}
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c:50]:
```suggestion
    if (clockwise) {
        tap_code(KC_VOLU);
    } else {
        tap_code(KC_VOLD);
    }
}
```

File: keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  if(clockwise)
+    tap_code(KC_VOLU);
+  else
+    tap_code(KC_VOLD);
+}
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/joypad/keymaps/via/keymap.c:50]:
```suggestion
    if (clockwise) {
        tap_code(KC_VOLU);
    } else {
        tap_code(KC_VOLD);
    }
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+[2] = LAYOUT_ortho_6x4(
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/joypad/keymaps/default/keymap.c:42]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**